src/uv-common.c

```diff
@@ -456,3 +456,70 @@ int uv_loop_configure(uv_loop_t* loop, uv_loop_option option, ...) {
 
   return err;
 }
+
+
+static uv_loop_t default_loop_struct;
+static uv_loop_t* default_loop_ptr;
+
+
+uv_loop_t* uv_default_loop(void) {
+  if (default_loop_ptr != NULL)
+    return default_loop_ptr;
+
+  if (uv_loop_init(&default_loop_struct))
+    return NULL;
+
+  default_loop_ptr = &default_loop_struct;
+  return default_loop_ptr;
+}
+
+
+uv_loop_t* uv_loop_new(void) {
+  uv_loop_t* loop;
+
+  loop = malloc(sizeof *loop);
+  if (loop == NULL) {
+    return NULL;
+  }
```

Comment:
I suppose that while you're here, you might as well touch up the style.  I believe `sizeof(*loop)` is slightly more common now and can you drop the braces?  Thanks!